File: examples/javascript/mpu9150.js
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */


var sensorObj = require('jsupm_mpu9150');

// Instantiate an MPU9105 on default I2C bus and address
var sensor = new sensorObj.MPU9150();

// Initialize the device with default values
sensor.init();

var x = new sensorObj.new_floatp();
var y = new sensorObj.new_floatp();
var z = new sensorObj.new_floatp();

// Output data every half second until interrupted
setInterval(function()
{
    sensor.update();
    
    sensor.getAccelerometer(x, y, z);
    console.log("Accelerometer: AX: " + sensorObj.floatp_value(x) + 
                " AY: " + sensorObj.floatp_value(y) + 
                " AZ: " + sensorObj.floatp_value(z));

    sensor.getGyroscope(x, y, z);
    console.log("Gyroscope:     GX: " + sensorObj.floatp_value(x) + 
                " AY: " + sensorObj.floatp_value(y) + 
                " AZ: " + sensorObj.floatp_value(z));
 
    sensor.getMagnetometer(x, y, z);
    console.log("Magnetometer:  MX: " + sensorObj.floatp_value(x) + 
                " MY: " + sensorObj.floatp_value(y) + 
                " MZ: " + sensorObj.floatp_value(z));

    console.log("Temperature:   " + sensor.getTemperature());
 
    console.log();

}, 500);

// exit on ^C
process.on('SIGINT', function()
{
    sensor = null;
    sensorObj.cleanUp();
    sensorObj = null;
    console.log("Exiting.");
    process.exit(0);
});

File: examples/c++/mpu9150.cxx
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <iostream>
#include <signal.h>

#include "mpu9150.hpp"
#include "upm_utilities.h"

using namespace std;

int shouldRun = true;

void
sig_handler(int signo)
{
    if (signo == SIGINT)
        shouldRun = false;
}

int
main(int argc, char** argv)
{
    signal(SIGINT, sig_handler);
    //! [Interesting]

    upm::MPU9150 sensor;

    sensor.init();

    while (shouldRun) {
        sensor.update();

        float x, y, z;

        sensor.getAccelerometer(&x, &y, &z);
        cout << "Accelerometer: ";
        cout << "AX: " << x << " AY: " << y << " AZ: " << z << endl;

        sensor.getGyroscope(&x, &y, &z);
        cout << "Gryoscope:     ";
        cout << "GX: " << x << " GY: " << y << " GZ: " << z << endl;

        sensor.getMagnetometer(&x, &y, &z);
        cout << "Magnetometer:  ";
        cout << "MX = " << x << " MY = " << y << " MZ = " << z << endl;

        cout << "Temperature:   " << sensor.getTemperature() << endl;
        cout << endl;

        upm_delay_us(500000);
    }

    //! [Interesting]

    cout << "Exiting..." << endl;

    return 0;
}
